[openmpi] Build failue

Package: openmpi[core]:x64-linux -> 4.1.6

**Host Environment**

- Host: x64-linux
- Compiler: GNU 13.2.1
-    vcpkg-tool version: 2023-03-29-664f8bb619b752430368d0f30a8289b761f5caba
    vcpkg-scripts version: 50bffcc62 2023-12-14 (5 hours ago)

**To Reproduce**

`vcpkg install openmpi`

**Failure logs**

Note, I had to shorten the logs. They were too long for github.

<details><summary>/opt/vcpkg/buildtrees/openmpi/config-x64-linux-dbg-err.log</summary>

```
configure: WARNING: -g has been added to CFLAGS (--enable-debug)
configure: WARNING:  -fno-strict-aliasing has been added to CFLAGS
configure: WARNING: -g has been added to CXXFLAGS (--enable-debug)
configure: WARNING:  -finline-functions has been added to CXXFLAGS
configure: WARNING: -g has been added to CXXFLAGS (--enable-debug)
configure: WARNING:  -finline-functions has been added to CXXFLAGS
configure: WARNING: MPI_REAL16 and MPI_COMPLEX32 support have been disabled
configure: WARNING: "-fvisibility=hidden" has been added to the hwloc CFLAGS
configure: WARNING: UCX support skipped because version 1.8.x was found, which has a known catastrophic issue.
configure: WARNING: Please upgrade to UCX version 1.9 or higher.
configure: WARNING: EXTERNAL PMIX SUPPORT REQUIRES USE OF EXTERNAL LIBEVENT
configure: WARNING: LIBRARY. THIS LIBRARY MUST POINT TO THE SAME ONE USED
configure: WARNING: TO BUILD PMIX OR ELSE UNPREDICTABLE BEHAVIOR MAY RESULT
configure: error: PLEASE CORRECT THE CONFIGURE COMMAND LINE AND REBUILD
```
</details>

**Additional context**

--- a/ports/openmpi/portfile.cmake
+++ b/ports/openmpi/portfile.cmake
@@ -26,6 +26,7 @@ vcpkg_configure_make(
         OPTIONS
             --with-hwloc=internal
             --with-libevent=internal
+            --with-pmix=internal
             --disable-mpi-fortran
         OPTIONS_DEBUG
             --enable-debug

--- a/ports/openmpi/vcpkg.json
+++ b/ports/openmpi/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "openmpi",
   "version": "4.1.6",
+  "port-version": 1,
   "description": "The Open MPI Project is an open source Message Passing Interface implementation that is developed and maintained by a consortium of academic, research, and industry partners. Open MPI is therefore able to combine the expertise, technologies, and resources from all across the High Performance Computing community in order to build the best MPI library available. Open MPI offers advantages for system and software vendors, application developers and computer science researchers.",
   "homepage": "https://www.open-mpi.org/",
   "supports": "!(windows | uwp)"

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6314,7 +6314,7 @@
     },
     "openmpi": {
       "baseline": "4.1.6",
-      "port-version": 0
+      "port-version": 1
     },
     "openmvg": {
       "baseline": "2.0",

--- a/versions/o-/openmpi.json
+++ b/versions/o-/openmpi.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "a63c7aeb8e42d6ebdac726767ec7a0ba46ddf86f",
+      "version": "4.1.6",
+      "port-version": 1
+    },
+    {
       "git-tree": "a35bc145d32795f1a71cbe596de040805c3af780",
       "version": "4.1.6",
       "port-version": 0

